modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -198,7 +202,12 @@ private RetryPolicyTemplate createRetryPolicyTemplate(RetryPolicyTemplate custom
   private void stopSource() throws MuleException {
     if (sourceAdapter != null) {
       try {
+        CoreEvent initialiserEvent = getInitialiserEvent(muleContext);
+        stopUsingConfiguration(initialiserEvent);
         sourceAdapter.stop();
+        if (usesDynamicConfiguration()) {
+          disposeSource();
```

Comment:
there's a bug here. If you dispose the source here, then it cannot be started again without invoking initialise() first. Add a test case in which :

1) the feature is tested
2) the source is stopped
3) the source is started
4) the feature is tested again and still works

Reply:
I will add this test, but I think this will work. When starting if the source adapter is null, it will recreate it. In the restart method we already do this :  stop() disposeSource() start().